src/platform/windows_winusb/enumeration.rs

```diff
@@ -124,13 +138,68 @@ pub fn probe_device(devinst: DevInst) -> Option<DeviceInfo> {
         protocol: info.device_desc.bDeviceProtocol,
         max_packet_size_0: info.device_desc.bMaxPacketSize0,
         speed: info.speed,
-        manufacturer_string: None,
+        manufacturer_string,
         product_string,
         serial_number,
         interfaces,
     })
 }
 
+pub fn probe_bus(devinst: DevInst) -> Option<BusInfo> {
+    let instance_id = devinst.get_property::<OsString>(DEVPKEY_Device_InstanceId)?;
+    // Skip non-root hubs; buses - ID will not parse
+    let (_device_version, _serial_number) = parse_root_hub_id(&instance_id)?;
+
+    debug!("Probing bus {instance_id:?}");
+
+    let parent_instance_id = devinst.get_property::<OsString>(DEVPKEY_Device_Parent)?;
+
+    let class_name = devinst
+        .get_property::<OsString>(DEVPKEY_Device_DeviceDesc)
+        .and_then(|s| s.into_string().ok());
+    let provider_class = devinst
+        .get_property::<OsString>(DEVPKEY_Device_Manufacturer)
+        .and_then(|s| s.into_string().ok());
+
+    let driver = get_driver_name(devinst);
+
+    let location_paths = devinst
+        .get_property::<Vec<OsString>>(DEVPKEY_Device_LocationPaths)
+        .unwrap_or_default();
+
+    let (bus_id, _) = location_paths
+        .iter()
+        .find_map(|p| parse_location_path(p))
+        .unwrap_or_default();
+
+    let pci_info = if let Some((vendor_id, device_id, revision, subsystem, _)) =
+        parse_host_controller_id(parent_instance_id.as_os_str())
+    {
+        Some(PciInfo {
+            instance_id: parent_instance_id,
+            vendor_id,
+            device_id,
+            revision: Some(revision as u16),
+            subsystem_vendor_id: Some((subsystem >> 16) as u16),
+            subsystem_device_id: Some((subsystem & 0xFFFF) as u16),
+        })
+    } else {
+        None
+    };
+
+    Some(BusInfo {
+        instance_id,
+        location_paths,
+        pci_info,
+        devinst,
+        driver: Some(driver).filter(|s| !s.is_empty()),
+        bus_id,
+        controller: None,
```

Comment:
It would be great if we could figure out a way to fill in the controller type on Windows.

Reply:
I considered parsing the device name on Windows; whether it has >= 3.0 (XHCI), 2.0 (EHCI) else OHCI. Should be accurate but it feels a bit dirty. I guess I could do this with a platform caveat covering this on Windows in the docs?

Reply:
It's a good point, the Host Controller contains a Service that seems to contain the controller type. I've updated to use this https://github.com/kevinmehall/nusb/pull/77/commits/366b2ab4b74d530c523ed2f6eb0ab749bf1023d3

Iterating over `GUID_DEVINTERFACE_USB_HUB` then getting the parent host controller like the Linux root hub exploration (rather than iterating over `GUID_DEVINTERFACE_USB_HUB`) still makes because a host controller can have more than one 'bus'. Iterating over the root hubs is the best way to build a `BusInfo` type I believe.